### PR TITLE
3PH part 1: Random drive-bys

### PR DESCRIPTION
**Pull request by kentonv**
(Split off from #2111)

Just some random fixes and cleanups before implementing starts in earnest.

### CODE REVIEW

File: c++/src/capnp/rpc.c++ [hunk 1]
```diff
@@ -1180,6 +1180,16 @@ private:
     Request<AnyPointer, AnyPointer> newCall(
         uint64_t interfaceId, uint16_t methodId, kj::Maybe<MessageSize> sizeHint,
         CallHints hints) override {
+      if (isResolved()) {
+        // Go directly to the resolved capability.
+        //
+        // If we don't do this now, then well return an RpcRequest which, as soon as send() is
```
Comment (ObsidianMinor) [inline c++/src/capnp/rpc.c++:1186]:
```suggestion
      // If we don't do this now, then we'll return an RpcRequest which, as soon as send() is
```

File: c++/src/capnp/rpc.c++ [hunk 2]
```diff
@@ -1190,6 +1200,11 @@ private:
 
     VoidPromiseAndPipeline call(uint64_t interfaceId, uint16_t methodId,
                                 kj::Own<CallContextHook>&& context, CallHints hints) override {
+      if (isResolved()) {
+        // Same as in newCall().
+        return cap->call(interfaceId, methodId, kj::mv(context), hints);
+      }
+
       receivedCall = true;
       return cap->call(interfaceId, methodId, kj::mv(context), hints);
```
Comment (ObsidianMinor) [inline c++/src/capnp/rpc.c++:1209]: Not sure how this is different? Just avoiding the `receivedCall` field setting.

Comment (kentonv) [reply c++/src/capnp/rpc.c++:1209]: Yeah you're right, it has no actual effect here, I was just trying to be consistent I guess.

Comment (kentonv) [reply c++/src/capnp/rpc.c++:1209]: (The setting of `receivedCall` doesn't matter post-resolve, since it only impacts what happens at resolve time.)

**Review by ObsidianMinor — APPROVED** [review]

**Pull request merged**